lib/openshift/machine_set.rb

```diff
@@ -45,5 +45,16 @@ def is_windows_machinesets?(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('spec', 'template', 'metadata', 'labels','machine.openshift.io/os-id') == "Windows"
     end 
+    def machineset_flavor(user: nil, cached: true, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      flavor = rr.dig('spec', 'template', 'spec', 'providerSpec', 'value', 'flavor')
+      if flavor.nil?
```

Comment:
@yanpzhan line 51 & 54 have the same condition check and it's assigning the variable `flavor` from different values of the Hash...the 2nd call wins always...are you sure this is that you want?

Reply:
Updated